src/v/model/record.h

```diff
@@ -420,6 +421,20 @@ struct record_batch_header {
         h.ctx.owner_shard = ss::this_shard_id();
         return h;
     }
+
+    /**
+     * Return true if the batch header is read from the zeroed fallocatated
+     * area.
+     *
+     * It is enough to compare first four fields of the batch to zero as valid
+     * batch type is newer equal to 0.
+     */
+    bool is_zeroed() const {
+        return header_crc == 0 && size_bytes == 0 && base_offset() == 0
```

Comment:
it seems to me that since it is valid for the header crc to be 0 that including it in the condition is irrelevant, right?

Reply:
it is partially relevant looking at the order of serialized fields in the header, they all should be 0

Reply:
oh I see its not a crc being calculated from the read data?